src/Microsoft.VisualStudio.ProjectSystem.Managed/GlobalSuppressions.cs

```diff
@@ -21,4 +21,6 @@
 [assembly: SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "<Pending>", Scope = "member", Target = "~M:Microsoft.VisualStudio.Threading.Tasks.SequentialTaskExecutor.ExecuteTask``1(System.Func{System.Threading.Tasks.Task{``0}})~System.Threading.Tasks.Task{``0}")]
 [assembly: SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "By design", Scope = "member", Target = "~M:Microsoft.VisualStudio.Threading.TaskResult.Null``1~System.Threading.Tasks.Task{``0}")]
 [assembly: SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "By design", Scope = "member", Target = "~M:Microsoft.VisualStudio.Threading.TaskResult.EmptyEnumerable``1~System.Threading.Tasks.Task{System.Collections.Generic.IEnumerable{``0}}")]
+[assembly: SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "<Pending>", Scope = "member", Target = "~M:Microsoft.VisualStudio.ProjectSystem.Debug.LaunchSettingsProviderExtensions.WaitForFirstSnapshot(Microsoft.VisualStudio.ProjectSystem.Debug.ILaunchSettingsProvider,System.Threading.CancellationToken)~System.Threading.Tasks.Task{Microsoft.VisualStudio.ProjectSystem.Debug.ILaunchSettings}")]
```

Comment:
>WaitForFirstSnapshot [](http://example.com/codeflow?start=224&length=20)

Should this be a new public interface if we want to encourage the correct use of the API? Or couldn't this extension actually have the async suffix since it is only used internally?

Reply:
The goal was for this to be an overload that pops up in completion alongside the existing one.

Adding a suffix to the new method would make it look like there was both a blocking and non blocking version, which is not true.

Also there are `When` method in the BCL that return take without `Async` suffixes.

Versioning the interface would require casting or changing the declared type in most usages. We may have to do that in future.